datumaro/components/media.py

```diff
@@ -564,16 +564,62 @@ class PointCloud(MediaElement):
 
     def __init__(
         self,
-        path: str,
-        extra_images: Optional[List[Image]] = None,
+        data: Union[bytes, Callable[[str], bytes], None] = None,
```

Comment:
I think that it has been not good architecture of Datumaro to take various input types (`data` and `path`) in one class constructor. This makes the constructor code for lazy loading so complicated. In addition, we already saw a bad side-effect from the Datumaro user who is unfamiliar with our product, such as storing raw images in Geti dataset-ie microservice. Therefore, I have had a plan in my mind to refactor all `Media`s to have own single input data type. For example,

**As-is**
```python
class Image:
    def __init__(path, data):
```

**To-be**
```python
class ImageFromFile:
    def __init__(path):

# Default is ImageFromFile
Image = ImageFromFile

class ImageFromData:
    def __init__(data):
```

Therefore, I think that it would be good to apply this kind of refactoring in `PointCloud` first, if you would like to add `data` in this PR.

Additionally, I think that it would be also good for introducing Apache Arrow file format. For example,

```python
class ImageFromArrow(ImageFromFile):
     def __init__(path, address_in_file):
```

Reply:
Glad you have such a plan. Let me change the implementation in a way you suggested.